### DISCUSSION TITLE
External Libraries Storage Requirements

### DISCUSSION THREAD

**Discussion by finickyFalcon5** [immich-app/immich#14232] [Q&A]
Hello,

I'm running Immich in a Debian 12 based LXC container inside Proxmox.
My images and videos are stored on a separate device (Debian server).
I mounted the server directory to a mount point on Proxmox, and then passed the directory to the container running Immich. I modified the Docker-Compose.yml file to include the external directory: "- /mnt/Pics_from_Server:/mnt/media/Pics_from_Server:ro"
I am able to add External Libraries with paths to the shared directory, but as Immich scans the External Libraries, the storage usage increases significantly - so far, the external libraries size is 320 GB, and the Immich storage usage is 90 GB (and growing).
I don't understand why the storage usage is so high. I thought that since the original photo/video files are stored externally, Immich would just create pointers and thumbnails to those files. I was expecting the Immich database to be very small.
How is sharing an external directory different than just uploading the files from the shared directory to Immich?
What am I misunderstanding about the storage requirements for External Libraries?
How can I estimate the required storage to allocate to Immich if I plan to index a shared directory with a size of X GB?

Thanks.

**Comment by mmomjian** [comment]
Immich creates a few various sizes of thumbnails for each image and an encoded video for most videos. You can take a look at the size of the thumbs/ and encoded-video/ folders to see how much space they are taking up. There is no hard rule as it depends on the file size of each original asset and the transcoding/thumbnail settings you have set. 90 GB for 320 GB seems a bit on the large side but if the external library is all images, especially lower resolution/smaller file size, it could be the case.

**Reply by finickyFalcon5** [reply]
Thanks.
I've added approx. 420 GB of external assets, and Immich is using 96.3 GB of storage.

**Comment by hettinger77** [comment]
I've noticed the same thing, the storage requirements were greatly over what I estimated.
